Mathlib/Algebra/Category/ModuleCat/Adjunctions.lean

```diff
@@ -160,44 +176,48 @@ instance : (free R).Monoidal :=
       μIso := μIso R
       μIso_hom_natural_left := fun {X Y} f X' ↦ by
         rw [← cancel_epi (μIso R X X').inv]
-        aesop
+        ext x
+        simp [μIso_inv_freeMk R, μIso_hom_freeMk_tmul_freeMk R]
```

Comment:
Do we understand why `simp [μIso_inv_freeMk, μIso_hom_freeMk_tmul_freeMk]` (or just `aesop`) does not work here, and if so how to fix it? The purpose of using a `def` rather than `abbrev` was precisely to improve automation. Reverting this design seems to break the part of automation which worked.

Reply:
I'm not really sure but I guess the simplest fix is phrasing those lemmas in terms of `Finsupp.single` instead (I've tried this in the latest commit)